part-1/docker-compose.yml

```diff
@@ -26,6 +23,13 @@ services:
                 "--notebook-dir=${JUPYTER_DIR}",
                 '--NotebookApp.token=${JUPYTER_TOKEN}'
             ]
+        deploy:
+            resources:
+                reservations:
+                    devices:
+                    - 'driver': 'nvidia'
+                      'count': 1
+                      'capabilities': ['gpu', 'utility']
```

Comment:
When specifying both 'gpu' and 'utility', CUDA is not available to PyTorch. Use only 'gpu'.